local_planner/src/nodes/waypoint_generator.cpp

```diff
@@ -339,6 +307,21 @@ void WaypointGenerator::adaptSpeed() {
   double delta_dist = since_update_sec * curr_vel_magnitude_;
   speed_ += delta_dist;
 
+  //break before goal
+  geometry_msgs::Point a;
+  a.x = std::abs(goal_.x - pose_.pose.position.x);
+  a.y = std::abs(goal_.y - pose_.pose.position.y);
+  a.z = std::abs(goal_.z - pose_.pose.position.z);
+  if (a.x < 3 * curr_vel_magnitude_ && a.y < 3 * curr_vel_magnitude_) {
```

Comment:
if either `a.x` or `a.y` are `>= 3.0 * curr_vel_magnitude` and  `<= 4.0 * curr_vel_magnitude`, `limit_speed_close_to_goal_` is not updated. Is that intended?

Reply:
yes, that is intended. It needs to have a hysteresis in there, because the distance at which we start to limit the velocity is velocity dependent. Meaning if we slow down the vehicle, the distance at which we should do so shrinks. This would lead to velocity discontiuities